polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx

```diff
@@ -82,11 +93,12 @@ export function MenuGroup({
       preferredAlignment="left"
       onClose={handleClose}
       hideOnPrint
+      preventCloseOnChildOverlayClick
```

Comment:
@knames just want to duplicate here the issue I found on the Product Detail Page with "More actions":
When all actions are collapsed under “More actions” on smaller screens, we are still losing a focus on the menu popover. Video attached.

https://user-images.githubusercontent.com/104942025/228057404-37f26363-ee45-4544-b357-11c5ac641bc8.mov



This makes it impossible for users to navigate using Tab.
And as @tatianau already mentioned 
`Tab or Shift + Tab - this is critical and in the scope of this PR as it's the basic keyboard navigation and keyboard users use this combination to navigate through the page, it must be supported. Getting the Tab and Shift + Tab to work with focus management is even more important than the Esc.`


This issue does not occur currently in production, so we have to figure out what Polaris changes might cause it to not introduce new issues in Admin

Reply:
I can't figure out how to reproduce this in stories. It seems to only be an issue for modals that are opened with `useModalContext` 

I don't really have a whole bunch more time to spend on this. If we can't figure this out soon, we can either ship what is currently here and just not attach `id` and `ref` to the secondaryActions where this is a known issue, or someone else will have to drive this home and figure out how to make the focus swap to these modals properly.

Reply:
Great investigation and summary @oksanashopify 🙌  Thank you so much for reproducing it in storybook, the admin can be pretty bloated, so it's great to see the repro steps in isolated environment like storybook to prove that the issue happens regardless of the consumer implementation but on its own and this seems to be the right layer to fix it.

That's right, with modals accessibility, we are moving the user’s focus as part of an interaction to alert them to new content, so that the keyboard users are in the right place, _not behind a modal window_, they are not stuck in some other part of the web application that isn’t active anymore. It's critical that the background layers are disabled when the modal is open so the focus only moves within the modal's interactive elements and not behind it on any other page content until the modal is closed, and if it's a regression to the existing modal experience in the admin - we need to fix it before merging 🙏

Reply:
nice work reproducing, seems it's doing weird shit with rollup. in this scenario, it pops one of the items out of the rollup and then breaks it.  looking into it

Reply:
with this commit 008fd5b160ea3f83c1f7dc5b0fa2ea931bd47861 , i think I fixed items popping out of more actions, which fixes the focus being stuck..

try it out with this playground (similar to yours, but I stuck `duplicate` under some other actions so it's always rolled up.


```ts
import React, {useCallback, useRef, useState} from 'react';
import {Page, Badge, Modal, TextContainer} from '@shopify/polaris';

export function Playground() {
  const duplicateActivatorRef = useRef<HTMLButtonElement>(null);
  const [active, setActive] = useState(true);
  const handleOpenDuplicateModal = useCallback(() => setActive(true), []);

  const handleClose = useCallback(() => {
    setActive(false);
  }, []);

  const duplicateProductModalMarkup = (
    <Modal
      activator={duplicateActivatorRef}
      open={active}
      onClose={handleClose}
      title="Reach more shoppers with Instagram product tags"
      primaryAction={{
        content: 'Add Instagram',
        onAction: handleClose,
      }}
      secondaryActions={[
        {
          content: 'Learn more',
          onAction: handleClose,
        },
      ]}
    >
      <Modal.Section>
        <TextContainer>
          <p>
            Use Instagram posts to share your products with millions of people.
            Let shoppers buy from your store without leaving Instagram.
          </p>
        </TextContainer>
      </Modal.Section>
    </Modal>
  );
  return (
    <>
      <Page
        breadcrumbs={[{content: 'Products', url: '#'}]}
        title="3/4 inch Leather pet collar"
        titleMetadata={<Badge status="success">Paid</Badge>}
        subtitle="Perfect for any pet"
        compactTitle
        primaryAction={{content: 'Save', disabled: true}}
        secondaryActions={[
          {
            content: 'Action 1',
            onAction: () => alert('View on your store action'),
          },
          {
            content: 'Action 2',
            onAction: () => alert('View on your store action'),
          },
          {
            content: 'Action 3',
            onAction: () => alert('View on your store action'),
          },
          {
            id: 'duplicate',
            content: 'Duplicate',
            accessibilityLabel: 'Secondary action label',
            onAction: () => handleOpenDuplicateModal(),
            ref: duplicateActivatorRef,
          },
        ]}
        actionGroups={[
          {
            title: 'More Actions',
            actions: [
              {
                content: 'Share on Facebook',
                accessibilityLabel: 'Individual action label',
                id: 'fb',
                onAction: () => handleOpenDuplicateModal(),
                ref: duplicateActivatorRef,
              },
            ],
          },
        ]}
        pagination={{
          hasPrevious: true,
          hasNext: true,
        }}
      />
      {duplicateProductModalMarkup}
    </>
  );
}

```

Reply:
https://screenshot.click/29-37-i7m8k-c0d3o.mp4

Reply:
@knames on the video you posted focus goes from the Modal not to the activator, but to the next action. Is it intended behaviour?

Reply:
also, secondary actions look broken for me in the playground and as well on Admin with the latest changes
<img width="988" alt="image" src="https://user-images.githubusercontent.com/104942025/228657434-3b1a273a-bb97-475f-9c8f-e1a29604b224.png">

Reply:
and going to the next activator was because I was using the same ref on two actions, that's fixed in my example

Reply:
wait it's broken again :/

Reply:
useMemo didn't work, by not showing actionsMarkup it seems to fix, but I'm assuming we want that

Reply:
https://screenshot.click/29-17-no6sr-k89yv.mp4 working!

Playground:

```ts
import React, {useCallback, useRef, useState} from 'react';
import {Page, Badge, Modal, TextContainer} from '@shopify/polaris';

export function Playground() {
  const duplicateActivatorRef = useRef<HTMLButtonElement>(null);
  const [active, setActive] = useState(true);
  const handleOpenDuplicateModal = useCallback(() => setActive(true), []);

  const handleClose = useCallback(() => {
    setActive(false);
  }, []);

  const duplicateProductModalMarkup = (
    <Modal
      activator={duplicateActivatorRef}
      open={active}
      onClose={handleClose}
      title="Reach more shoppers with Instagram product tags"
      primaryAction={{
        content: 'Add Instagram',
        onAction: handleClose,
      }}
      secondaryActions={[
        {
          content: 'Learn more',
          onAction: handleClose,
        },
      ]}
    >
      <Modal.Section>
        <TextContainer>
          <p>
            Use Instagram posts to share your products with millions of people.
            Let shoppers buy from your store without leaving Instagram.
          </p>
        </TextContainer>
      </Modal.Section>
    </Modal>
  );
  return (
    <>
      <Page
        breadcrumbs={[{content: 'Products', url: '#'}]}
        title="3/4 inch Leather pet collar"
        titleMetadata={<Badge status="success">Paid</Badge>}
        subtitle="Perfect for any pet"
        compactTitle
        primaryAction={{content: 'Save', disabled: true}}
        secondaryActions={[
          {
            content: 'Action 1',
            onAction: () => alert('View on your store action'),
          },
          {
            content: 'Action 2',
            onAction: () => alert('View on your store action'),
          },
          {
            content: 'Action 3',
            onAction: () => alert('View on your store action'),
          },
          {
            id: 'duplicate',
            content: 'Duplicate',
            accessibilityLabel: 'Secondary action label',
            onAction: () => handleOpenDuplicateModal(),
            ref: duplicateActivatorRef,
          },
        ]}
        actionGroups={[
          {
            title: 'More Actions',
            actions: [
              {
                content: 'Share on Facebook',
                accessibilityLabel: 'Individual action label',
              },
            ],
          },
        ]}
        pagination={{
          hasPrevious: true,
          hasNext: true,
        }}
      />
      {duplicateProductModalMarkup}
    </>
  );
}

```

Reply:
cutting popover work